spec/models/room_spec.rb

```diff
@@ -1,7 +1,27 @@
 require 'rails_helper'
 
 RSpec.describe Room, type: :model do
+  let!(:aspen) { Hotel.create!(name: "Tyrolean Inn", location: "Aspen") }
+  let!(:bum) { aspen.rooms.create!(rate: 100, suite: "Ski Bum") }
+  let!(:steve) { Guest.create!(name: "Steve", nights: 2) }
+  let!(:steve) { Guest.create!(name: "Steve", nights: 2) }
+  let!(:stevie) { Guest.create!(name: "Stevie", nights: 2) }
+
+  before do
+    GuestRoom.create!(guest: steve, room: bum)
+    GuestRoom.create!(guest: steve, room: bum)
+    GuestRoom.create!(guest: stevie, room: bum)
+  end
+
   describe 'relationships' do
     it {should belong_to :hotel}
+    it {should have_many(:guests).through(:guest_rooms)}
+  end
+
+  describe 'Instance Methods' do
+    it '#guest_count' do
+      expect(bum.guest_count).to eq(2)
+      expect(bum.guest_count).not_to eq(3)
+    end
   end
 end
```

Comment:
👍 

Great choice making this an instance method on the `Room` model. I think that is the easiest approach for what is being asked. The only thing I would say to add is another assertion to verify that the value, `2`, isn't hard-coded in the `guest_count` method.

Remember that when we are following TDD, it says that we should write the "smallest amount of code in order to get the test to pass"... in this case, the smallest amount of code would be to just hard-code the value `2` in the method.

```ruby
def guest_count
  2
end
```

Just by looking at the test, there's no way to tell is the logic is hard-coded or if the method is actually performing some logic. In order to make it obvious I like to add another assertion with an updated value.

```ruby
expect(bum.guest_count).to eq(2)
expect(bum.guest_count).not_to eq(3)

let!(:sam) { Guest.create!(name: "Sam", nights: 2) }
GuestRoom.create!(guest: sam, room: bum)

expect(bum.guest_count).to eq(3)
```

With this change, we've seen the value of `guest_count` change from 2 to 3, expressing that the logic is not hard-coded and updates as the data changes.